test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -268,16 +272,40 @@ double UcxContext::connect_timeout() const
     return _connect_timeout;
 }
 
+int UcxContext::is_timeout_elapsed(struct timeval *tv_later,
+                                   struct timeval *tv_prior, double timeout)
+{
+    struct timeval elapsed;
+
+    timersub(tv_later, tv_prior, &elapsed);
+    if (elapsed.tv_sec + (elapsed.tv_usec * 1e-6) > timeout) {
+        return 1;
+    }
+    return 0;
+}
+
 void UcxContext::progress_conn_requests()
 {
     while (!_conn_requests.empty()) {
-        UcxConnection *conn = new UcxConnection(*this, get_next_conn_id());
-        if (conn->accept(_conn_requests.front())) {
+        UcxConnection *conn     = new UcxConnection(*this, get_next_conn_id());
+        conn_req_t conn_request = _conn_requests.front();
+
+        struct timeval tv_current;
+        gettimeofday(&tv_current, NULL);
```

Comment:
put this into is_timeout_elapsed as well

Reply:
do we always want to compare with current time? why not leave it as an arg?

Reply:
yes, to unite common code